### PR TITLE
Updating copyright header for all .py files

### PR DESCRIPTION
**Pull request by ksbeattie**
## Summary/Motivation:

Copyright header needs updating for year and consistency across all files

## Changes proposed in this PR:
- Adding `.addheader.yml` config and `copyright_header.txt` files for running addheader command
- Modifying nearly all files by running `addheader -c .addheader.yml` adding/updating the copyright header

## Reviewer's checklist / merge requirements:
- [ ] The head branch (i.e. the "source" of the changes) is not the `main` branch on the PR author's fork
- [ ] Documentation
- [ ] Tests
- [ ] Diagnostic tests for models

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: copyright_header.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+“ProMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
```
Comment (MarcusHolly) [inline copyright_header.txt:1]:
```suggestion
“PrOMMiS” was produced under the DOE Process Optimization and Modeling for Minerals Sustainability
```

**Review by bpaul4 — APPROVED** [review]
Looks good, I'm curious how it knows to delete the old headers in files that have them?

Also, we should probably add a bullet to the "Reviewer's checklist / merge requirements" to check that all Python files have the correct header.

File: copyright_header.txt (hunk 1 above)
Comment (ksbeattie) [reply copyright_header.txt]: Good catch!  I've updated it and re-ran `addheader` so all the files have changed - again.

**Comment by ksbeattie** [comment]
> Looks good, I'm curious how it knows to delete the old headers in files that have them?
> 
> Also, we should probably add a bullet to the "Reviewer's checklist / merge requirements" to check that all Python files have the correct header.

This is all done using Dan's [addheader](https://github.com/IDAES/addheader) script - it's pretty slick.

**Review by MarcusHolly — APPROVED** [review]
LGTM

**Pull request merged**